pyuvdata/tests/test_uvflag.py

```diff
@@ -0,0 +1,1280 @@
+if uvversion.git_hash is not '':
+    pyuvdata_version_str += ('  Git origin: ' + uvversion.git_origin
+                             + '.  Git hash: ' + uvversion.git_hash
+                             + '.  Git branch: ' + uvversion.git_branch
+                             + '.  Git description: ' + uvversion.git_description + '.')
+
+
+def test_init_UVData():
+    uv = UVData()
+    uv.read_miriad(test_d_file)
+    uvf = UVFlag(uv, history='I made a UVFlag object', label='test')
+    nt.assert_true(uvf.metric_array.shape == uv.flag_array.shape)
+    nt.assert_true(np.all(uvf.metric_array == 0))
+    nt.assert_true(uvf.weights_array.shape == uv.flag_array.shape)
+    nt.assert_true(np.all(uvf.weights_array == 1))
+    nt.assert_true(uvf.type == 'baseline')
+    nt.assert_true(uvf.mode == 'metric')
+    nt.assert_true(np.all(uvf.time_array == uv.time_array))
+    nt.assert_true(np.all(uvf.lst_array == uv.lst_array))
+    nt.assert_true(np.all(uvf.freq_array == uv.freq_array[0]))
+    nt.assert_true(np.all(uvf.polarization_array == uv.polarization_array))
+    nt.assert_true(np.all(uvf.baseline_array == uv.baseline_array))
+    nt.assert_true(np.all(uvf.ant_1_array == uv.ant_1_array))
+    nt.assert_true(np.all(uvf.ant_2_array == uv.ant_2_array))
+    nt.assert_true('I made a UVFlag object' in uvf.history)
+    nt.assert_true('Flag object with type "baseline"' in uvf.history)
+    nt.assert_true(pyuvdata_version_str in uvf.history)
+    nt.assert_true(uvf.label == 'test')
+
+
+def test_init_UVData_copy_flags():
+    uv = UVData()
+    uv.read_miriad(test_d_file)
+    uvf = uvtest.checkWarnings(UVFlag, [uv], {'copy_flags': True, 'mode': 'metric'},
+                               nwarnings=1, message='Copying flags to type=="baseline"')
+    nt.assert_false(hasattr(uvf, 'metric_array'))  # Should be flag due to copy flags
+    nt.assert_true(np.array_equal(uvf.flag_array, uv.flag_array))
+    nt.assert_true(uvf.weights_array.shape == uv.flag_array.shape)
+    nt.assert_true(np.all(uvf.weights_array == 1))
+    nt.assert_true(uvf.type == 'baseline')
+    nt.assert_true(uvf.mode == 'flag')
+    nt.assert_true(np.all(uvf.time_array == uv.time_array))
+    nt.assert_true(np.all(uvf.lst_array == uv.lst_array))
+    nt.assert_true(np.all(uvf.freq_array == uv.freq_array[0]))
+    nt.assert_true(np.all(uvf.polarization_array == uv.polarization_array))
+    nt.assert_true(np.all(uvf.baseline_array == uv.baseline_array))
+    nt.assert_true(np.all(uvf.ant_1_array == uv.ant_1_array))
+    nt.assert_true(np.all(uvf.ant_2_array == uv.ant_2_array))
+    nt.assert_true('Flag object with type "baseline"' in uvf.history)
+    nt.assert_true(pyuvdata_version_str in uvf.history)
+
+
+@uvtest.skipIf_no_hera_cal
+def test_init_HERAData():
+    from hera_cal.io import HERAData
+    uv = UVData()
+    uv.read_miriad(test_d_file)
+    uvf1 = UVFlag(uv)
+    hd = HERAData(test_d_file, filetype='miriad')
+    hd.read()
+    uvf2 = UVFlag(hd)
+    nt.assert_equal(uvf1, uvf2)
```

Comment:
What utility are you testing here that requires hera_cal? It seems like all of this can be done with just pyuvdata...

Reply:
This will be moved back to hera_qm.